### ISSUE TITLE
oc:build takes ages to perform

### ISSUE THREAD

**Issue by sunix** [eclipse/jkube#1506]
### Description
with this project https://github.com/sebastienblanc/antennas-front
it takes ages to do this part....?
![Selection_657](https://user-images.githubusercontent.com/650571/169242820-20df087c-dc13-436a-b829-ff142b782507.png)

![Selection_656](https://user-images.githubusercontent.com/650571/169242849-e6c181aa-e9fa-4f1d-86a4-d714813d0b16.png)

### Info

* Eclipse JKube version : 1.7.0
* Maven version (`mvn -v`) :
```

```
* Kubernetes / Red Hat OpenShift setup and version :

* If it's a bug, how to reproduce :

* If it's a feature request, what is your use case :

* Sample Reproducer Project : *[GitHub Clone URL]*

**Comment by manusa** [comment]
Hi @sunix 
did you manage to figure out why it was taking so long in your case? do you have an environment where we can reproduce this so we can profile it?

**Comment by sunix** [comment]
@manusa no unfortunately the environment has gone.

**Comment by rohanKanojia** [comment]
I tried cloning the project again and doing `oc:build`. These were my observations

- Red Hat Developer Sandbox : `mvn oc:build` took `01:28 min`
- CRC : `mvn oc:build` took `54.276 sec`

I still feel this could be a slow cluster or a network issue.

**Comment by sunix** [comment]
OK I think we can close this issue, It was probably a slow network: I made a comparison (in the same condition) by triggering the build with `oc` to compare the files upload:
with `oc`: 79s
with jkube: ~69s

So to compare I first made a first `mvn clean package oc:build` that is trigering a Build but also creating a BuildConfig in openshift. Then I am measuring an upload with `oc` (having that `code-with-quarkus-s2i` is the name of the BuildConfig)
```sh
start_time=$SECONDS; oc start-build code-with-quarkus-s2i --from-dir=. ; echo $(( SECONDS - start_time ))
Uploading directory "." as binary input for the build ...
...............
Uploading finished
build.build.openshift.io/code-with-quarkus-s2i-11 started
79
```
Basically it took 79s

Waiting for the build to finish, I then run the command `mvn oc:build`. I start a stopwatcher when is logged `[INFO] oc: Starting Build code-with-quarkus-s2i` and stop the stopwatcher once it is starting to log again `[INFO] oc: Starting Build code-with-quarkus-s2i`. I used my phone as a stopwatcher as I just wanted to have a rough idea of the time: 69s. It is even faster than oc.
![Selection_679](https://user-images.githubusercontent.com/650571/177506504-87b3174b-af80-4dd7-8612-77a64bf3eb2c.png)

So I am closing this issue as we don't see major differences and assuming that this was due to a slow network at the venue of the hackathon.
